src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1567,12 +1566,10 @@ note: if you're sure you want to do this, please open an issue as to why. In the
             cmd.arg("--coverage-dump-path").arg(coverage_dump);
         }
 
-        if mode == "run-coverage" {
-            // The demangler doesn't need the current compiler, so we can avoid
-            // unnecessary rebuilds by using the bootstrap compiler instead.
+        if mode == "run-make" || mode == "run-coverage" {
             let rust_demangler = builder
                 .ensure(tool::RustDemangler {
-                    compiler: compiler.with_stage(0),
+                    compiler,
```

Comment:
rust-demangler is intentionally on stage 0 to avoid unnecessary recompilations: https://github.com/rust-lang/rust/pull/117197